src/main/catalog-pusher.ts

```diff
@@ -24,10 +24,17 @@ import { broadcastMessage } from "../common/ipc";
 import type { CatalogEntityRegistry } from "./catalog";
 import "../common/catalog-entities/kubernetes-cluster";
 import { toJS } from "../common/utils";
+import { debounce } from "lodash";
+import type { CatalogEntity } from "../common/catalog";
+
+
+const broadcaster = debounce((items: CatalogEntity[]) => {
+  broadcastMessage("catalog:items", items);
+}, 1_000, { trailing: true });
 
 export function pushCatalogToRenderer(catalog: CatalogEntityRegistry) {
   return reaction(() => toJS(catalog.items), (items) => {
-    broadcastMessage("catalog:items", items);
+    broadcaster(items);
   }, {
     fireImmediately: true,
```

Comment:
Probably debounce can be reached with `delay: 1000`

```suggestion
    fireImmediately: true,
    delay: 1000,
```